### PR TITLE
joinmarket: bump secp256k1 to version used upstream

### PR DESCRIPTION
**Pull request by nixbitcoin**
Also incorporate improvements from upstream nixpkgs expression.

GPG verified upstream commit and repo checksum.

### CODE REVIEW

**Review by erikarvstedt — DISMISSED** [review]
ACK becde23a691d8126b047bcab005528dbcfd3edf0

File: pkgs/secp256k1/default.nix [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
 
-  configureFlags = ["--enable-module-recovery" "--disable-jni" "--enable-experimental" "--enable-module-ecdh" "--enable-benchmark=no" ];
+  configureFlags = [
+    "--enable-benchmark=no"
+    "--enable-exhaustive-tests=no"
+    "--enable-experimental"
+    "--enable-module-ecdh"
+    "--enable-module-recovery"
+    "--enable-module-schnorrsig"
+    "--enable-tests=yes"
```
Comment (jonasnick) [inline pkgs/secp256k1/default.nix:28]: nit: tests are enabled by default

Comment (nixbitcoin) [reply pkgs/secp256k1/default.nix]: Fixed in 6629e9a66f1b4aea5feb30a89c0be0ff5c6dd1a0

**Review by erikarvstedt — APPROVED** [review]
ACK 6629e9a66f1b4aea5feb30a89c0be0ff5c6dd1a0

**Review by jonasnick — COMMENTED** [review]
ACK 6629e9a66f1b4aea5feb30a89c0be0ff5c6dd1a0

**Pull request merged**